apps/wallet/src/background/accounts/zklogin/ZkLoginAccount.ts

```diff
@@ -148,6 +175,25 @@ export class ZkLoginAccount
 			createdAt: Date.now(),
 			claimName,
 		};
+
+		// By default, always import the legacy address. If the legacy and
+		// non-legacy addresses differ, only import the non-legacy address if it
+		// has already been used.
+		ret.push({
+			...accountData,
+			address: legacyAddress,
+		});
+		if (normalizeSuiAddress(legacyAddress) !== normalizeSuiAddress(nonLegacyAddress)) {
+			if (await hasTransactionHistory(nonLegacyAddress)) {
+				ret.push({
+					...accountData,
+					address: nonLegacyAddress,
+					nickname: null,
```

Comment:
Probably should still set this, but just with `(address 2)` appended per the discussion